### ISSUE TITLE
Cursor not on the right position when dragging item.

### ISSUE THREAD

**Issue by KhaldiAmer** [atlassian/react-beautiful-dnd#1748]
### Expected behavior 

The cursor is on the element being dragged.

### Actual behavior

The cursor is away from the element which makes it hard sometimes to drag to the wanted position.

### Steps to reproduce
1- Resize draggable elements  when onBeforeCapture is triggered. 
2- Drag second element to the last element and notice the cursor position while dragging.

### Suggested solution?

None for now.

### What version of `React` are you using?

16.11

### What version of `react-beautiful-dnd` are you running?

12.2

### What browser are you using?

Version 79.0.3945.130  (64 bits)

### Demo
https://codesandbox.io/s/vertical-list-kgs03?fontsize=14&hidenavigation=1&theme=dark

**Comment by alexvcasillas** [comment]
Hey there, I believe `onBeforeCapture` should be asynchronous (should accept a done callback) to support asynchronous evaluations on components

**Comment by holmberd** [comment]
@alexvcasillas true, not much time to do anything worthwhile. Alternatively if there was a way to set custom dimensions on drag to avoid placing a callback in the critical path. 

Reference:

https://github.com/atlassian/react-beautiful-dnd/blob/c8c86601b3666b8236ff46d56daeedaf9c41e566/src/state/middleware/responders/publisher.js#L89-L105

https://github.com/atlassian/react-beautiful-dnd/blob/c8c86601b3666b8236ff46d56daeedaf9c41e566/src/state/middleware/responders/responders-middleware.js#L26-L36

https://github.com/atlassian/react-beautiful-dnd/blob/c8c86601b3666b8236ff46d56daeedaf9c41e566/src/state/middleware/lift.js#L38-L55

**Comment by gabriel-os-sousa** [comment]
@KhaldiAmer did you get any solution?

**Comment by KhaldiAmer** [comment]
> @KhaldiAmer did you get any solution?

Nothing yet.

**Comment by angelitolm** [comment]
I have the same problem

**Comment by Kaishley** [comment]
were you guys able to solve this?

**Comment by JohnHuang-CT** [comment]
I also ran into this problem, does anyone already have a solution?

**Comment by bailinx** [comment]
I have the same problem...

**Comment by SerejaCrystalys** [comment]
Same..

**Comment by andrewpeterson99** [comment]
Running into the same issue. Best solution for now might just be reworking GUI to avoid this kind of interaction imo.
